error

got this error:

web-1  | INFO:     172.18.0.1:40382 - "GET / HTTP/1.1" 200 OK
web-1  | INFO:     172.18.0.1:40382 - "GET /favicon.ico HTTP/1.1" 404 Not Found
web-1  | 2025-11-13 11:53:57,043 [INFO] app.main - Generated Google OAuth URL for state 0044f884-25d4-4add-b5ca-a2797ab60282
web-1  | INFO:     172.18.0.1:47844 - "GET /auth/google HTTP/1.1" 302 Found
web-1  | 2025-11-13 11:54:03,576 [INFO] app.main - Received OAuth callback for state 0044f884-25d4-4add-b5ca-a2797ab60282
web-1  | 2025-11-13 11:54:03,809 [INFO] app.main - OAuth token exchange succeeded for state 0044f884-25d4-4add-b5ca-a2797ab60282
web-1  | 2025-11-13 11:54:03,927 [INFO] app.main - Building Google Ads client for user 2
web-1  | 2025-11-13 11:54:04,169 [INFO] app.main - Syncing accessible customers for user 2
web-1  | 2025-11-13 11:54:05,507 [INFO] app.google_ads_client - Google Ads returned 9 accessible customers in 0.57s
web-1  | 2025-11-13 11:54:05,507 [INFO] app.main - Retrieved 9 accessible customer resource names
web-1  | 2025-11-13 11:54:05,546 [INFO] app.main - Selected login customer ID 8174598300 for user 2
web-1  | 2025-11-13 11:54:05,546 [INFO] app.main - Customer sync completed in 1.38s
web-1  | INFO:     172.18.0.1:47856 - "GET /oauth/callback?state=0044f884-25d4-4add-b5ca-a2797ab60282&code=4/0Ab32j90rayHHxje3so56Cef4aOGL-886h8z4VhAq7yLEq4DMyv7O1batLcvCqVht7k9h-Q&scope=https://www.googleapis.com/auth/adwords HTTP/1.1" 302 Found
web-1  | 2025-11-13 11:54:05,669 [INFO] app.main - Listing campaigns for user 2
web-1  | 2025-11-13 11:54:06,544 [INFO] app.google_ads_client - Fetched 0 campaigns for customer 8174598300 in 0.57s
web-1  | 2025-11-13 11:54:06,544 [INFO] app.main - Fetched 0 campaigns for customer 8174598300 in 0.60s
web-1  | 2025-11-13 11:54:06,571 [INFO] app.main - Persisted 0 campaigns for customer 8174598300 in 0.03s
web-1  | INFO:     172.18.0.1:47856 - "GET /campaigns HTTP/1.1" 200 OK
web-1  | 2025-11-13 11:54:18,648 [INFO] app.main - Listing campaigns for user 2
web-1  | Request made: ClientCustomerId: 5363645115, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: azDUErkcMuQjxkS60efm6w, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 11:54:19,355 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "azDUErkcMuQjxkS60efm6w"
web-1  | , 'azDUErkcMuQjxkS60efm6w')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 414, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "azDUErkcMuQjxkS60efm6w"
web-1  | , 'azDUErkcMuQjxkS60efm6w')
web-1  | Request made: ClientCustomerId: 5363645115, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: ww5HAXO2XEFWS6qU2NUKSA, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 11:54:21,672 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "ww5HAXO2XEFWS6qU2NUKSA"
web-1  | , 'ww5HAXO2XEFWS6qU2NUKSA')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 414, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "ww5HAXO2XEFWS6qU2NUKSA"
web-1  | , 'ww5HAXO2XEFWS6qU2NUKSA')
web-1  | Request made: ClientCustomerId: 5363645115, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: mfaSlznqtMTzoHgZ1r_vuQ, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 11:54:26,085 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "mfaSlznqtMTzoHgZ1r_vuQ"
web-1  | , 'mfaSlznqtMTzoHgZ1r_vuQ')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 414, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "mfaSlznqtMTzoHgZ1r_vuQ"
web-1  | , 'mfaSlznqtMTzoHgZ1r_vuQ')
web-1  | Request made: ClientCustomerId: 5363645115, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: yvwFzfumRZe8IjRyaAF2sw, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 11:54:34,455 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "yvwFzfumRZe8IjRyaAF2sw"
web-1  | , 'yvwFzfumRZe8IjRyaAF2sw')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 414, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "yvwFzfumRZe8IjRyaAF2sw"
web-1  | , 'yvwFzfumRZe8IjRyaAF2sw')
web-1  | Request made: ClientCustomerId: 5363645115, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: 2eOwgCeq5qyUDp-LV22-dw, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 11:54:50,842 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.250.109.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "2eOwgCeq5qyUDp-LV22-dw"
web-1  | , '2eOwgCeq5qyUDp-LV22-dw')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 414, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call

@claude now got this error:
web-1  | INFO:     172.18.0.1:45816 - "GET / HTTP/1.1" 200 OK
web-1  | INFO:     172.18.0.1:45816 - "GET /favicon.ico HTTP/1.1" 404 Not Found
web-1  | 2025-11-13 12:26:17,370 [INFO] app.main - Generated Google OAuth URL for state 3c78b230-4d7e-4278-81e0-e9824d55b5cd
web-1  | INFO:     172.18.0.1:45816 - "GET /auth/google HTTP/1.1" 302 Found
web-1  | 2025-11-13 12:26:28,062 [INFO] app.main - Received OAuth callback for state 3c78b230-4d7e-4278-81e0-e9824d55b5cd
web-1  | 2025-11-13 12:26:28,342 [INFO] app.main - OAuth token exchange succeeded for state 3c78b230-4d7e-4278-81e0-e9824d55b5cd
web-1  | 2025-11-13 12:26:28,408 [INFO] app.main - Building Google Ads client for user 2
web-1  | 2025-11-13 12:26:28,657 [INFO] app.main - Syncing accessible customers for user 2
web-1  | 2025-11-13 12:26:30,055 [INFO] app.google_ads_client - Google Ads returned 9 accessible customers in 0.61s
web-1  | 2025-11-13 12:26:30,055 [INFO] app.main - Retrieved 9 accessible customer resource names
web-1  | 2025-11-13 12:26:30,093 [INFO] app.main - Selected login customer ID 8174598300 for user 2
web-1  | 2025-11-13 12:26:30,093 [INFO] app.main - Customer sync completed in 1.44s
web-1  | INFO:     172.18.0.1:45818 - "GET /oauth/callback?state=3c78b230-4d7e-4278-81e0-e9824d55b5cd&code=4/0Ab32j91Zr51NvBtLICD_nhbdHyFNI3f-8M_NRB5-6uJRkHhvy8Hfi1hCZ8obKUnZl1xNXw&scope=https://www.googleapis.com/auth/adwords HTTP/1.1" 302 Found
web-1  | 2025-11-13 12:26:30,200 [INFO] app.main - Listing campaigns for user 2
web-1  | 2025-11-13 12:26:30,896 [INFO] app.google_ads_client - Fetched 0 campaigns for customer 8174598300 in 0.41s
web-1  | 2025-11-13 12:26:30,896 [INFO] app.main - Fetched 0 campaigns for customer 8174598300 in 0.42s
web-1  | 2025-11-13 12:26:30,916 [INFO] app.main - Persisted 0 campaigns for customer 8174598300 in 0.02s
web-1  | INFO:     172.18.0.1:45818 - "GET /campaigns HTTP/1.1" 200 OK
web-1  | 2025-11-13 12:26:55,715 [INFO] app.main - Listing campaigns for user 2
web-1  | Request made: ClientCustomerId: 9600473470, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: UtECCZ2s9kFd_Vm0J2wEWA, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 12:26:56,374 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "UtECCZ2s9kFd_Vm0J2wEWA"
web-1  | , 'UtECCZ2s9kFd_Vm0J2wEWA')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 443, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "UtECCZ2s9kFd_Vm0J2wEWA"
web-1  | , 'UtECCZ2s9kFd_Vm0J2wEWA')
web-1  | Request made: ClientCustomerId: 9600473470, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: VXmsYi3VMnOBerb92lTCXw, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 12:26:58,729 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "VXmsYi3VMnOBerb92lTCXw"
web-1  | , 'VXmsYi3VMnOBerb92lTCXw')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 443, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "VXmsYi3VMnOBerb92lTCXw"
web-1  | , 'VXmsYi3VMnOBerb92lTCXw')
web-1  | Request made: ClientCustomerId: 9600473470, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: 7otd45k76NkiF-9r6wCMBA, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 12:27:03,055 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "7otd45k76NkiF-9r6wCMBA"
web-1  | , '7otd45k76NkiF-9r6wCMBA')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 443, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "7otd45k76NkiF-9r6wCMBA"
web-1  | , '7otd45k76NkiF-9r6wCMBA')
web-1  | Request made: ClientCustomerId: 9600473470, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: 5NPCGzEy7wOrL_dKimO29w, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 12:27:11,443 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "5NPCGzEy7wOrL_dKimO29w"
web-1  | , '5NPCGzEy7wOrL_dKimO29w')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 443, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_status:7, grpc_message:"The caller does not have permission"}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "5NPCGzEy7wOrL_dKimO29w"
web-1  | , '5NPCGzEy7wOrL_dKimO29w')
web-1  | Request made: ClientCustomerId: 9600473470, Host: googleads.googleapis.com, Method: /google.ads.googleads.v22.services.GoogleAdsService/Search, RequestId: lq6gWKm-7WYBqp7HALQNRA, IsFault: True, FaultMessage: User doesn't have permission to access customer. Note: If you're accessing a client customer, the manager's customer id must be set in the 'login-customer-id' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
web-1  | 2025-11-13 12:27:27,873 [ERROR] app.google_ads_client - Failed to list campaigns: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "lq6gWKm-7WYBqp7HALQNRA"
web-1  | , 'lq6gWKm-7WYBqp7HALQNRA')
web-1  | Traceback (most recent call last):
web-1  |   File "/app/app/google_ads_client.py", line 443, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "lq6gWKm-7WYBqp7HALQNRA"
web-1  | , 'lq6gWKm-7WYBqp7HALQNRA')
web-1  | INFO:     172.18.0.1:44862 - "GET /campaigns?customer_id=9600473470 HTTP/1.1" 500 Internal Server Error
web-1  | ERROR:    Exception in ASGI application
web-1  | Traceback (most recent call last):
web-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 399, in run_asgi
web-1  |     result = await app(  # type: ignore[func-returns-value]
web-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
web-1  |     return await self.app(scope, receive, send)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
web-1  |     await super().__call__(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
web-1  |     await self.middleware_stack(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
web-1  |     raise exc
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
web-1  |     await self.app(scope, receive, _send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/sessions.py", line 85, in __call__
web-1  |     await self.app(scope, receive, send_wrapper)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
web-1  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
web-1  |     raise exc
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
web-1  |     await app(scope, receive, sender)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 756, in __call__
web-1  |     await self.middleware_stack(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 776, in app
web-1  |     await route.handle(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 297, in handle
web-1  |     await self.app(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 77, in app
web-1  |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
web-1  |     raise exc
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
web-1  |     await app(scope, receive, sender)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 72, in app
web-1  |     response = await func(request)
web-1  |                ^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 278, in app
web-1  |     raw_response = await run_endpoint_function(
web-1  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
web-1  |     return await dependant.call(**values)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/app/app/main.py", line 413, in list_campaigns
web-1  |     campaigns = service.list_campaigns(
web-1  |                 ^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/tenacity/__init__.py", line 289, in wrapped_f
web-1  |     return self(f, *args, **kw)
web-1  |            ^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/tenacity/__init__.py", line 379, in __call__
web-1  |     do = self.iter(retry_state=retry_state)
web-1  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/tenacity/__init__.py", line 325, in iter
web-1  |     raise retry_exc.reraise()
web-1  |           ^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/tenacity/__init__.py", line 158, in reraise
web-1  |     raise self.last_attempt.result()
web-1  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 449, in result
web-1  |     return self.__get_result()
web-1  |            ^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/tenacity/__init__.py", line 382, in __call__
web-1  |     result = fn(*args, **kwargs)
web-1  |              ^^^^^^^^^^^^^^^^^^^
web-1  |   File "/app/app/google_ads_client.py", line 443, in list_campaigns
web-1  |     response = ga_service.search(
web-1  |                ^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/v22/services/services/google_ads_service/client.py", line 4614, in search
web-1  |     response = rpc(
web-1  |                ^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
web-1  |     return wrapped_func(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callable
web-1  |     return callable_(*args, **kwargs)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 276, in __call__
web-1  |     response, ignored_call = self._with_call(
web-1  |                              ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 331, in _with_call
web-1  |     return call.result(), call
web-1  |            ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 170, in result
web-1  |     raise self._exception
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 314, in continuation
web-1  |     response, call = self._thunk(new_method).with_call(
web-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 342, in with_call
web-1  |     return self._with_call(
web-1  |            ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/grpc/_interceptor.py", line 328, in _with_call
web-1  |     call = self._interceptor.intercept_unary_unary(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 99, in intercept_unary_unary
web-1  |     self._handle_grpc_failure(response)
web-1  |   File "/usr/local/lib/python3.11/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 71, in _handle_grpc_failure
web-1  |     raise self._get_error_from_response(response)
web-1  | google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, <_InactiveRpcError of RPC that terminated with:
web-1  |        status = StatusCode.PERMISSION_DENIED
web-1  |        details = "The caller does not have permission"
web-1  |        debug_error_string = "UNKNOWN:Error received from peer ipv4:142.251.98.95:443 {grpc_message:"The caller does not have permission", grpc_status:7}"
web-1  | >, errors {
web-1  |   error_code {
web-1  |     authorization_error: USER_PERMISSION_DENIED
web-1  |   }
web-1  |   message: "User doesn\'t have permission to access customer. Note: If you\'re accessing a client customer, the manager\'s customer id must be set in the \'login-customer-id\' header. See https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid"
web-1  | }
web-1  | request_id: "lq6gWKm-7WYBqp7HALQNRA"
web-1  | , 'lq6gWKm-7WYBqp7HALQNRA')
Gracefully stopping... (press Ctrl+C again to force)
[+] Stopping 1/1
 ✔ Container super-duper-umbrella-web-1  Stopped                                                                   0.4s


C:\Users\andre\super-duper-umbrella>docker compose build --no-cache web
[+] Building 84.9s (14/14) FINISHED
 => [internal] load local bake definitions                                                                         0.0s
 => => reading from stdin 408B                                                                                     0.0s
 => [internal] load build definition from Dockerfile                                                               0.0s
 => => transferring dockerfile: 460B                                                                               0.0s
 => [internal] load metadata for docker.io/library/python:3.11-slim                                                1.3s
 => [internal] load .dockerignore                                                                                  0.0s
 => => transferring context: 2B                                                                                    0.0s
 => [1/7] FROM docker.io/library/python:3.11-slim@sha256:e4676722fba839e2e5cdb844a52262b43e90e56dbd55b7ad953ee361  0.0s
 => => resolve docker.io/library/python:3.11-slim@sha256:e4676722fba839e2e5cdb844a52262b43e90e56dbd55b7ad953ee361  0.0s
 => [internal] load build context                                                                                  0.0s
 => => transferring context: 164.69kB                                                                              0.0s
 => CACHED [2/7] WORKDIR /app                                                                                      0.0s
 => [3/7] RUN apt-get update && apt-get install -y --no-install-recommends build-essential curl     && rm -rf /v  20.8s
 => [4/7] COPY requirements.txt ./                                                                                 0.0s
 => [5/7] RUN pip install --no-cache-dir -r requirements.txt                                                      40.8s
 => [6/7] COPY app ./app                                                                                           0.1s
 => [7/7] COPY README.md ./README.md                                                                               0.0s
 => exporting to image                                                                                            21.3s
 => => exporting layers                                                                                           14.0s
 => => exporting manifest sha256:a9b4dfcdc40fb10937575fe6b8ac461fb23c8d842ad801774fef8ed526baa2da                  0.0s
 => => exporting config sha256:c4888328f89abeef320c559e8e3c96ba659b70c9f2a9a85429f6418db14b0fcf                    0.0s
 => => exporting attestation manifest sha256:224415ffa5087aa49b02d31fb9bf166b194c04502eaa70e07fd8f07edfdd7eba      0.0s
 => => exporting manifest list sha256:b1bb5f7b328ae907c9ab965b79022362174a71ce4c135bb8208726228cbb3165             0.0s
 => => naming to docker.io/library/super-duper-umbrella-web:latest                                                 0.0s
 => => unpacking to docker.io/library/super-duper-umbrella-web:latest                                              7.1s
 => resolving provenance for metadata file                                                                         0.0s
[+] Building 1/1
 ✔ web  Built                                                                                                      0.0s

C:\Users\andre\super-duper-umbrella>docker compose up
[+] Running 2/2
 ✔ Network super-duper-umbrella_default  Created                                                                   0.0s
 ✔ Container super-duper-umbrella-web-1  Created                                                                   5.7s
Attaching to web-1
web-1  | INFO:     Will watch for changes in these directories: ['/app']
web-1  | INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
web-1  | INFO:     Started reloader process [1] using WatchFiles
web-1  | 2025-11-13 13:26:33,053 [INFO] app.logging_setup - Application logs initialised at logs/latest-run.log
web-1  | 2025-11-13 13:26:33,053 [INFO] app.main - Logging configured. File output: logs/latest-run.log
web-1  | INFO:     Started server process [8]
web-1  | INFO:     Waiting for application startup.
web-1  | 2025-11-13 13:26:33,069 [INFO] app.main - Starting application initialisation
web-1  | 2025-11-13 13:26:33,097 [INFO] app.main - Database initialised in 0.03s
web-1  | INFO:     Application startup complete.
web-1  | INFO:     172.18.0.1:54254 - "GET / HTTP/1.1" 200 OK
web-1  | INFO:     172.18.0.1:54254 - "GET /favicon.ico HTTP/1.1" 404 Not Found
web-1  | 2025-11-13 13:27:16,344 [INFO] app.main - Generated Google OAuth URL for state 9e4c8c4a-15a8-425a-b412-97e4bdd0cb3e
web-1  | INFO:     172.18.0.1:54254 - "GET /auth/google HTTP/1.1" 302 Found
web-1  | 2025-11-13 13:27:20,905 [INFO] app.main - Received OAuth callback for state 9e4c8c4a-15a8-425a-b412-97e4bdd0cb3e
web-1  | 2025-11-13 13:27:21,190 [INFO] app.main - OAuth token exchange succeeded for state 9e4c8c4a-15a8-425a-b412-97e4bdd0cb3e
web-1  | 2025-11-13 13:27:21,212 [INFO] app.main - Building Google Ads client for user 2
web-1  | 2025-11-13 13:27:21,423 [INFO] app.main - Syncing accessible customers for user 2
web-1  | 2025-11-13 13:27:22,653 [INFO] app.google_ads_client - Google Ads returned 9 accessible customers in 0.62s
web-1  | 2025-11-13 13:27:22,653 [INFO] app.main - Retrieved 9 accessible customer resource names
web-1  | INFO:     172.18.0.1:54254 - "GET /oauth/callback?state=9e4c8c4a-15a8-425a-b412-97e4bdd0cb3e&code=4/0Ab32j935APAW0DC5MONrGt_Zw7sxVf8lyUgxXVwZ0K6gbgSpktX-Ag8Dg9NN44EluceNLw&scope=https://www.googleapis.com/auth/adwords HTTP/1.1" 500 Internal Server Error
web-1  | ERROR:    Exception in ASGI application
web-1  | Traceback (most recent call last):
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
web-1  |     self.dialect.do_execute(
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 924, in do_execute
web-1  |     cursor.execute(statement, parameters)
web-1  | sqlite3.OperationalError: no such column: customers.manager_customer_id
web-1  |
web-1  | The above exception was the direct cause of the following exception:
web-1  |
web-1  | Traceback (most recent call last):
web-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 399, in run_asgi
web-1  |     result = await app(  # type: ignore[func-returns-value]
web-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
web-1  |     return await self.app(scope, receive, send)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
web-1  |     await super().__call__(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
web-1  |     await self.middleware_stack(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
web-1  |     raise exc
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
web-1  |     await self.app(scope, receive, _send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/sessions.py", line 85, in __call__
web-1  |     await self.app(scope, receive, send_wrapper)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
web-1  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
web-1  |     raise exc
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
web-1  |     await app(scope, receive, sender)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 756, in __call__
web-1  |     await self.middleware_stack(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 776, in app
web-1  |     await route.handle(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 297, in handle
web-1  |     await self.app(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 77, in app
web-1  |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
web-1  |     raise exc
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
web-1  |     await app(scope, receive, sender)
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 72, in app
web-1  |     response = await func(request)
web-1  |                ^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 278, in app
web-1  |     raw_response = await run_endpoint_function(
web-1  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 193, in run_endpoint_function
web-1  |     return await run_in_threadpool(dependant.call, **values)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/concurrency.py", line 42, in run_in_threadpool
web-1  |     return await anyio.to_thread.run_sync(func, *args)
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/anyio/to_thread.py", line 56, in run_sync
web-1  |     return await get_async_backend().run_sync_in_worker_thread(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2485, in run_sync_in_worker_thread
web-1  |     return await future
web-1  |            ^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 976, in run
web-1  |     result = context.run(func, *args)
web-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/app/app/main.py", line 216, in oauth_callback
web-1  |     sync_customers(session, user, service)
web-1  |   File "/app/app/main.py", line 304, in sync_customers
web-1  |     existing = session.execute(
web-1  |                ^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2351, in execute
web-1  |     return self._execute_internal(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2236, in _execute_internal
web-1  |     result: Result[Any] = compile_state_cls.orm_execute_statement(
web-1  |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/context.py", line 293, in orm_execute_statement
web-1  |     result = conn.execute(
web-1  |              ^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1418, in execute
web-1  |     return meth(
web-1  |            ^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/sql/elements.py", line 515, in _execute_on_connection
web-1  |     return connection._execute_clauseelement(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1640, in _execute_clauseelement
web-1  |     ret = self._execute_context(
web-1  |           ^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1846, in _execute_context
web-1  |     return self._exec_single_context(
web-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1986, in _exec_single_context
web-1  |     self._handle_dbapi_exception(
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 2353, in _handle_dbapi_exception
web-1  |     raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
web-1  |     self.dialect.do_execute(
web-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 924, in do_execute
web-1  |     cursor.execute(statement, parameters)
web-1  | sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such column: customers.manager_customer_id
web-1  | [SQL: SELECT customers.id, customers.user_id, customers.resource_name, customers.customer_id, customers.descriptive_name, customers.currency_code, customers.time_zone, customers.manager_customer_id, customers.created_at, customers.updated_at
web-1  | FROM customers
web-1  | WHERE customers.resource_name = ?]
web-1  | [parameters: ('customers/2879789166',)]
web-1  | (Background on this error at: https://sqlalche.me/e/20/e3q8)

--- a/app/db.py
+++ b/app/db.py
@@ -6,7 +6,7 @@ from contextlib import contextmanager
 from pathlib import Path
 from typing import Any, Generator
 
-from sqlalchemy import create_engine
+from sqlalchemy import create_engine, text
 from sqlalchemy.engine import make_url
 from sqlalchemy.orm import DeclarativeBase, Session, sessionmaker
 
@@ -49,12 +49,44 @@ engine = create_engine(DATABASE_URL, connect_args=connect_args, future=True)
 SessionLocal = sessionmaker(bind=engine, autocommit=False, autoflush=False, class_=Session, future=True)
 
 
+def _migrate_schema() -> None:
+    """Apply schema migrations for existing databases.
+
+    This function handles adding new columns to existing tables that were created
+    before the column was added to the model. It's a lightweight alternative to
+    full Alembic migrations for simple schema changes.
+    """
+    import logging
+
+    logger = logging.getLogger(__name__)
+
+    # Only perform migrations for SQLite databases
+    if not url.drivername.startswith("sqlite"):
+        return
+
+    with engine.connect() as conn:
+        # Check if manager_customer_id column exists in customers table
+        result = conn.execute(
+            text("SELECT COUNT(*) FROM pragma_table_info('customers') WHERE name='manager_customer_id'")
+        )
+        column_exists = result.scalar() > 0
+
+        if not column_exists:
+            logger.info("Adding manager_customer_id column to customers table")
+            conn.execute(text("ALTER TABLE customers ADD COLUMN manager_customer_id VARCHAR(32)"))
+            conn.commit()
+            logger.info("Successfully added manager_customer_id column")
+
+
 def init_db() -> None:
-    """Create all tables."""
+    """Create all tables and apply necessary migrations."""
     # Import models within the function to avoid circular imports.
     from . import models  # noqa: F401
 
     Base.metadata.create_all(bind=engine)
+
+    # Apply schema migrations for existing databases
+    _migrate_schema()
 
 
 @contextmanager

